ArduCopter/GCS_Mavlink.cpp

```diff
@@ -570,6 +570,48 @@ bool GCS_MAVLINK_Copter::set_home(const Location& loc, bool _lock) {
     return copter.set_home(loc, _lock);
 }
 
+MAV_RESULT GCS_MAVLINK_Copter::handle_command_int_do_reposition(const mavlink_command_int_t &packet) {
+    const bool change_modes = ((int32_t)packet.param2 & MAV_DO_REPOSITION_FLAGS_CHANGE_MODE) == MAV_DO_REPOSITION_FLAGS_CHANGE_MODE;
+    if (!copter.flightmode->in_guided_mode() && !change_modes) {
+        return MAV_RESULT_DENIED;
+    }
+
+    // sanity check location
+    if (!check_latlng(packet.x, packet.y)) {
+        return MAV_RESULT_DENIED;
+    }
+
+    Location requested_position {};
+    requested_position.lat = packet.x;
+    requested_position.lng = packet.y;
+
+    // check the floating representation for overflow of altitude
+    if (fabsf(packet.z * 100.0f) >= 0x7fffff) {
```

Comment:
I've never seen this check before, by the way we have, "#define LOCATION_ALT_MAX_M  83000" in AP_Common/Location.h which is a very similar number but slightly lower and this apparently is the limit of what we can fit into the Location class.

Reply:
I had copied it elsewhere out of plane originally, but it's since disappeared, and I don't disagree that using the define is both far clearer, and a tighter range is better, swapped.